src/lib/core/components/visualizations/implementations/BoxplotVisualization.tsx

```diff
@@ -144,6 +144,16 @@ function BoxplotViz(props: Props) {
 
   const findEntityAndVariable = useFindEntityAndVariable(entities);
 
+  // outputEntity for OutputEntityTitle's outputEntity prop and outputEntityId at getRequestParams
+  const outputEntity = useMemo(() => {
+    const outputEntityVariableName =
+      dataElementDependencyOrder != null &&
+      dataElementDependencyOrder[0] === 'yAxisVariable'
+        ? vizConfig.yAxisVariable
+        : vizConfig.xAxisVariable;
+    return findEntityAndVariable(outputEntityVariableName)?.entity;
+  }, [dataElementDependencyOrder, vizConfig, findEntityAndVariable]);
```

Comment:
I think this is the repetitive code you were referring to. Maybe it can be generalized into a hook like this:

```typescript
function useFindOutputEntity(
  dataElementDependencyOrder?: string[],
  dataElementVariables: Record<string, VariableDescriptor | undefined>,
  defaultVariableName?: string,
  entities: StudyEntity[]
): EntityAndVariable | undefined {
  const findEntityAndVariable = useFindEntityAndVariable(entities);
  return useMemo(
    () => {
      const variableName = (dataElementDependencyOrder == null || dataElementDependencyOrder.length === 0)
        ? defaultVariableName
        : dataelementDependencyOrder[0];
      const variable = dataElementVariables[variableName];
      return findEntityAndVariable(variable)?.entity;
    },
    [dataElementDependencyOrder, dataElementVariables, defaultValue, entities ]
  );
}
```

It could be used like this:
```typescript
const outputEntity = useFindOutputEntity(
  dataElementDependencyOrder,
  vizConfig,
  'xAxisVariable',
  entities
);
```

Keep in mind that this is untested code, so some modifications may be necessary.

Reply:
Thank you for suggestion! 👍 After battling ts issues (haha), I could update it